libr/debug/p/native/linux/linux_debug.c

```diff
@@ -97,6 +97,85 @@ int linux_handle_signals(RDebug *dbg, int tid) {
 		// siginfo.si_code -> HWBKPT, USER, KERNEL or WHAT
 		// TODO: DO MORE RDEBUGREASON HERE
 		switch (dbg->reason.signum) {
+		case SIGABRT: // 6 / SIGIOT // SIGABRT
+			dbg->reason.type = R_DEBUG_REASON_SIGNAL;
+			dbg->reason.signum = R_SIGNAL_SIGABRT;
+			break;
+		case SIGALRM:
+			dbg->reason.type = R_DEBUG_REASON_SIGNAL;
+			dbg->reason.signum = R_SIGNAL_SIGALRM;
+			break;
+		case SIGBUS:
+			dbg->reason.type = R_DEBUG_REASON_SIGNAL;
+			dbg->reason.signum = R_SIGNAL_SIGBUS;
+			break;
+		case SIGCHLD:
+			dbg->reason.type = R_DEBUG_REASON_SIGNAL;
+			dbg->reason.signum = R_SIGNAL_SIGCHLD;
+			break;
+		case SIGCONT:
+			dbg->reason.type = R_DEBUG_REASON_SIGNAL;
+			dbg->reason.signum = R_SIGNAL_SIGCONT;
+			break;
+#if __sparc__
+		case SIGEMT:
+			dbg->reason.type = R_DEBUG_REASON_SIGNAL;
+			dbg->reason.signum = R_SIGNAL_SIGEMT;
+			break;
+#endif
+		case SIGFPE:
+			dbg->reason.type = R_DEBUG_REASON_SIGNAL;
+			dbg->reason.signum = R_DEBUG_REASON_FPU;
+			break;
+		case SIGHUP:
+			dbg->reason.type = R_DEBUG_REASON_SIGNAL;
+			dbg->reason.signum = R_SIGNAL_SIGHUP;
+			break;
+		case SIGILL:
+			dbg->reason.type = R_DEBUG_REASON_SIGNAL;
+			dbg->reason.signum = R_SIGNAL_SIGILL;
+			break;
+		case SIGPWR:
+			break;
+		case SIGINT:
+			dbg->reason.type = R_DEBUG_REASON_USERSUSP;
+			break;
+		case SIGIO:
+			break;
+		case SIGKILL:
+			dbg->reason.type = R_DEBUG_REASON_SIGNAL;
+			dbg->reason.signum = R_SIGNAL_SIGKILL;
+			break;
+		case SIGPIPE:
+			break;
+		case SIGPROF:
+			dbg->reason.type = R_DEBUG_REASON_SIGNAL;
+			dbg->reason.signum = R_SIGNAL_SIGPROF;
+			break;
+		case SIGQUIT:
+			dbg->reason.type = R_DEBUG_REASON_SIGNAL;
+			dbg->reason.signum = R_SIGNAL_SIGQUIT;
+			break;
+		case SIGSEGV:
+			dbg->reason.type = R_DEBUG_REASON_SIGNAL;
+			dbg->reason.signum = R_SIGNAL_SIGSEGV;
+			break;
+		case SIGSTKFLT:
+			dbg->reason.type = R_DEBUG_REASON_SIGNAL;
+			dbg->reason.signum = R_SIGNAL_SIGSTKFLT;
+			break;
+		case SIGSTOP:
+			dbg->reason.type = R_DEBUG_REASON_SIGNAL;
+			dbg->reason.signum = R_SIGNAL_SIGSTOP;
+			break;
+		case SIGSYS:
+			dbg->reason.type = R_DEBUG_REASON_SIGNAL;
```

Comment:
Just set signal as type at the begining and set the other values for the cases that are considered stop ressons. The code will be much shorter